pymc3/distributions/mixture.py

```diff
@@ -139,6 +142,150 @@ def random_choice(*args, **kwargs):
             return np.squeeze(comp_samples[w_samples])
 
 
+class ZeroInflatedPoisson(Mixture):
+    R"""
+    Zero-inflated Poisson log-likelihood.
+
+    Often used to model the number of events occurring in a fixed period
+    of time when the times at which events occur are independent.
+
+    .. math::
+
+        f(x \mid \psi, \theta) = \left\{ \begin{array}{l}
+            (1-\psi) + \psi e^{-\theta}, \text{if } x = 0 \\
+            \psi \frac{e^{-\theta}\theta^x}{x!}, \text{if } x=1,2,3,\ldots
+            \end{array} \right.
+
+    ========  ==========================
+    Support   :math:`x \in \mathbb{N}_0`
+    Mean      :math:`\psi\theta`
+    Variance  :math:`\theta + \frac{1-\psi}{\psi}\theta^2`
+    ========  ==========================
+
+    Parameters
+    ----------
+    psi : float
+        Expected proportion of Poisson variates (0 < psi < 1)
+    theta : float
+        Expected number of occurrences during the given interval
+        (theta >= 0).
+
+
+    """
+
+    def __init__(self, psi, theta, *args, **kwargs):
+        self.theta = theta = tt.as_tensor_variable(theta)
```

Comment:
you should also cast `psi` into tensor (same below), although might not necessary after #2233 is merge